### PR TITLE
Migrated to python3.6 and django 1.11

### PR DESCRIPTION
**Pull request by anjali-dhanuka**
# Description
Project ported to django 1.11 and python 3.6.

Fixes  https://github.com/systers/vms/issues/557

# Type of Change:
- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

# Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by anjali-dhanuka** [comment]
@tonythomas01 @tapasweni-pathak Porting to django 1.11 complete. Please review.

File: vms/administrator/urls.py [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
 
-urlpatterns = patterns(
-    '',
-    url(r'^report/$', GenerateReportView.as_view(), name='report'),
-    url(r'^settings/$', views.settings, name='settings'),
-)
+urlpatterns = [
+                       url(r'^report/$', GenerateReportView.as_view(), name='report'),
+                       url(r'^settings/$', views.settings, name='settings'),
+                       ]
```
Comment (tonythomas01) [inline vms/administrator/urls.py]: over intended here.

File: vms/authentication/urls.py [hunk 2]
```diff
@@ -1,19 +1,13 @@
...
     url(r'^$', views.index, name='index'),
-    url(r'^login/$',
-        anonymous_required(auth_views.login),
-        {'template_name': 'authentication/login.html'},
-        name='login_process'),
-    url(r'^logout/$',
-        auth_views.logout, {'template_name': 'home/home.html'},
-        name='logout_process'),
-)
+    url(r'^login/$', anonymous_required(auth_views.login), {'template_name': 'authentication/login.html'}, name='login_process'),
```
Comment (tonythomas01) [inline vms/authentication/urls.py]: too long line

File: vms/shift/urls.py [hunk 3]
```diff
@@ -1,18 +1,13 @@
 # Django
-from django.conf.urls import patterns, url
+from django.conf.urls import url
 
 # local Django
 from shift import views
 from shift.views import AddHoursView, AddHoursManagerView, ShiftCreateView, ShiftDeleteView, ClearHoursView, EditHoursView, ShiftUpdateView, ClearHoursManager, EditHoursManagerView, JobListView, ShiftListView, ManageVolunteerShiftView, ViewHoursView, VolunteerSearchView
```
Comment (tonythomas01) [inline vms/shift/urls.py]: too long line. Lets better keep to <= 79

File: vms/shift/urls.py [hunk 4]
```diff
@@ -34,28 +29,12 @@
...
-        views.view_volunteer_shifts,
-        name='view_volunteer_shifts'),
-    url(r'^volunteer_search/$',
-        VolunteerSearchView.as_view(),
-        name='volunteer_search'),
-)
+    url(r'^list_shifts/(?P<job_id>\d+)$', ShiftListView.as_view(), name='list_shifts'),
+    url(r'^view_shift/(?P<shift_id>\d+)$', views.view_volunteers, name='view_volunteers'),
+    url(r'^list_shifts_sign_up/(?P<job_id>\d+)/(?P<volunteer_id>\d+)$', views.list_shifts_sign_up, name='list_shifts_sign_up'),
+    url(r'^manage_volunteer_shifts/(?P<volunteer_id>\d+)$', ManageVolunteerShiftView.as_view(), name='manage_volunteer_shifts'),
```
Comment (tonythomas01) [inline vms/shift/urls.py]: too long lines

File: vms/vms/settings.py [hunk 5]
```diff
@@ -68,14 +68,23 @@
...
+        'OPTIONS': {
+            'context_processors': [
+                'django.template.context_processors.debug',
+                'django.template.context_processors.request',
+                'django.contrib.auth.context_processors.auth',
+                'django.contrib.messages.context_processors.messages',
+                ],
+            'debug': DEBUG
+            },
+         },
```
Comment (tonythomas01) [inline vms/vms/settings.py]: badly intended `}`

File: vms/vms/urls.py [hunk 6]
```diff
@@ -21,4 +21,4 @@
         include('registration.urls', namespace='registration')),
     url(r'^shift/', include('shift.urls', namespace='shift')),
     url(r'^volunteer/', include('volunteer.urls', namespace="volunteer")),
```
Comment (tonythomas01) [inline vms/vms/urls.py]: use `''` instead of `""` for namespace= please.

**Comment by anjali-dhanuka** [comment]
@tonythomas01 @tapasweni-pathak Please review. Proper formatting is also implemented.

**Comment by anjali-dhanuka** [comment]
@tapasweni-pathak vms uses psql 9.2 which is not supported by django 1.11

**Comment by anjali-dhanuka** [comment]
@tonythomas01 @tapasweni-pathak Done with the changes from my side. Still the test database is not forming. Can u suggest me anything I can do.

File: vms/manage.py [hunk 7]
```diff
@@ -2,9 +2,22 @@
...
-
+    try:
+        from django.core.management import execute_from_command_line
+    except ImportError:
+        # The above import may fail for some other reason. Ensure that the
+        # issue is really that Django is missing to avoid masking other
+        # exceptions on Python 2.
+        try:
+            import django
+        except ImportError:
```
Comment (tonythomas01) [inline vms/manage.py]: I dont know why you should have this file change though. Please revert all changes in this file.

**Comment by tonythomas01** [comment]
`. Still the test database is not forming` - did you try setting up travis locally and try to execute things ? You might get some clue then. https://stackoverflow.com/questions/21053657/how-to-run-travis-ci-locally

**Comment by anjali-dhanuka** [comment]
@tonythomas01 I ran travis commands locally. The test database is not formed. Django_migrations table is empty too in the test db. Make Migrations and migrate command does run properly, but the same doesn't happen while creating test database.

File: vms/manage.py (hunk 7 above)
Comment (anjali-dhanuka) [reply vms/manage.py]: @tonythomas01 manage.py is restored. Please review.

File: vms/administrator/models.py [hunk 8]
```diff
@@ -62,7 +62,7 @@ class Administrator(models.Model):
     organization = models.ForeignKey(Organization, null=True)
     # EmailField automatically checks if email address is a valid format
     email = models.EmailField(max_length=45, unique=True)
-    user = models.OneToOneField(User)
+    user = models.OneToOneField(settings.AUTH_USER_MODEL)
```
Comment (tonythomas01) [inline vms/administrator/models.py]: undef ?

**Comment by anjali-dhanuka** [comment]
@tonythomas01 I have made the change but still the tests are failing.
@sa7i  @tonythomas01 @abhishekarya286 @murad @ramitsahwney  @tapasweni-pathak Can u please help me with the tests.

**Comment by abhishekarya286** [comment]
Will look into this today. Sorry for the delay.

**Comment by MeepyMay** [comment]
@anjali-dhanuka Create a new PR to the gsoc18-code branch.

**Comment by ayushgarg1804** [comment]
Closing it right now due to #679 in another branch

**Pull request closed without merging**